forui/lib/src/widgets/tooltip.dart

```diff
@@ -0,0 +1,396 @@
+        onEnter: (_) => _enter(),
+        onExit: (_) => _exit(),
+        // We have to use a Listener as GestureDetector's arena implementation allows only 1 gesture to win. It is
+        // problematic if the child is a button. See https://github.com/flutter/flutter/issues/92103.
+        child: Listener(
+          onPointerDown: (_) => _exit(),
+          child: child,
+        ),
+      );
+    }
+
+    // TODO: haptic feedback.
+    if (widget.behavior == FToolTipBehavior.longPress || widget.behavior == FToolTipBehavior.hoverAndLongPress) {
+      child = GestureDetector(
+        onLongPressStart: (_) async {
+          _fencingToken = UniqueKey();
+          await _controller.show();
+        },
+        onLongPressEnd: (_) async {
+          final fencingToken = _fencingToken = UniqueKey();
+          await Future.delayed(widget.longPressExitDuration);
+
+          if (fencingToken == _fencingToken) {
+            await _controller.hide();
+          }
+        },
+        child: child,
+      );
+    }
+
+    return FPortal(
+      controller: _controller._overlay,
+      targetAnchor: widget.childAnchor,
+      followerAnchor: widget.tipAnchor,
+      shift: widget.shift,
+      followerBuilder: (context) => Semantics(
+        container: true,
+        child: FadeTransition(
+          opacity: _controller._fade,
+          child: ScaleTransition(
+            scale: _controller._scale,
+            child: Padding(
+              padding: style.margin,
+              child: DecoratedBox(
+                decoration: style.decoration,
+                child: Padding(
+                  padding: style.padding,
+                  child: DefaultTextStyle(
+                    style: style.textStyle,
+                    child: widget.tipBuilder(context, style, null),
+                  ),
+                ),
+              ),
+            ),
+          ),
+        ),
+      ),
+      child: child,
+    );
+  }
+
+  Future<void> _enter() async {
+    final fencingToken = _fencingToken = UniqueKey();
+    await Future.delayed(widget.hoverEnterDuration);
+
+    if (fencingToken == _fencingToken) {
+      await _controller.show();
+    }
+  }
+
+  Future<void> _exit() async {
+    final fencingToken = _fencingToken = UniqueKey();
+    await Future.delayed(widget.hoverExitDuration);
+
+    if (fencingToken == _fencingToken) {
+      await _controller.hide();
+    }
+  }
+
+  @override
+  void dispose() {
+    if (widget.controller == null) {
+      _controller.dispose();
+    }
+    super.dispose();
+  }
+}
+
+/// A [FTooltip]'s style.
+final class FTooltipStyle with Diagnosticable {
+  /// The tooltip's default shadow in [FTooltipStyle.inherit].
+  static const shadow = [
+    BoxShadow(
+      color: Color(0x1a000000),
+      offset: Offset(0, 4),
+      blurRadius: 6,
+      spreadRadius: -1,
+    ),
+    BoxShadow(
+      color: Color(0x1a000000),
+      offset: Offset(0, 2),
+      blurRadius: 4,
+      spreadRadius: -2,
+    ),
+  ];
```

Comment:
Is this intended? They wouldn't be able to change the values of static variables.

Reply:
Yup it's intended. It's a constant value that they can use in their box decoration.